WIP: Update of the README document

quietlychris opened the pull request:
As the working group has grown a little bit, I figured it might be valuable to start thinking about a review of the layout and content of the Github organization. This pull request has a first draft of suggested changes to the README, which functionally acts as the WG's homepage for developers. Feedback and changes are requested!

quietlychris committed:
Initial draft of README update

deg4uss3r commented on README.md:
```diff
@@ -5,29 +5,31 @@
 - https://rust-ml.zulipchat.com
 
-### Organization
+We also host a video chats every two weeks. Meeting minutes can be found in the [Meetings](https://github.com/rust-ml/Meetings) repository.
```
I would also mention meetings are normally announced in the Zulip

quietlychris replied:
Good suggestion, will do

deg4uss3r commented on README.md:
```diff
@@ -5,29 +5,31 @@
+    - The Rust-ML group actively develops and maintiners linfa, a Rust machine learning framework/library, which fills a similar role as Python's scikit-learn
 
-We welcome interest from the community, so please feel free to participate!
+More importantly, Rust's machine learning community is filled of experimentation! Take a look at some of the projects currently being worked on by members on the working group's [Task Board](https://github.com/rust-ml/wg/blob/master/Task_Board.md), or take a look at some of the project's being discussed in our Zulip chat. 
```
> More importantly, Rust's machine learning community is filled of experimentation! 

More importantly, Rust's machine learning community is filled with experimentation!

quietlychris replied:
Ugh, sorry about that. **with** experimentation, will change

deg4uss3r commented on README.md:
```diff
@@ -5,29 +5,31 @@
+    - The Rust-ML group actively develops and maintiners linfa, a Rust machine learning framework/library, which fills a similar role as Python's scikit-learn
 
-We welcome interest from the community, so please feel free to participate!
+More importantly, Rust's machine learning community is filled of experimentation! Take a look at some of the projects currently being worked on by members on the working group's [Task Board](https://github.com/rust-ml/wg/blob/master/Task_Board.md), or take a look at some of the project's being discussed in our Zulip chat. 
```
link Zulip chat again

quietlychris replied:
Will do, can't have too many links to the chat...

deg4uss3r commented on README.md:
```diff
@@ -5,29 +5,31 @@
-The Rust-ML group has a Zulip channel. Feel free to take a look and say hi!
+- Create spaces where people interested in doing machine learning in Rust can meet, ask questions, and discuss their work
+- Help organize support for Rust libraries, tools, and projects in the machine learning domain
+- In rare instances, to assist with or assume the maintainership of projects important to the wider community
```
Maybe reword? I think I would like to foster Rust help to ML experts, the way this is worded might drive them off slightly?

xd009642 replied:
What about an additional bullet point, something like `- Create a mature production ready machine learning ecosystem`

quietlychris replied:
This is definitely the weakest area of the change, and I think it's worth some additional discussion. During the video meeting today, @deg4uss3r mentioned something along the lines of helping people with machine learning become familiar with Rust, and enabling people familiar with Rust to do machine learning. 

@xd009642 This is definitely an area for discussion about WG goals. For example, the Embedded Group very clearly directly develops crates and tooling, whereas the GameDev WG explicity does not build things like games/game engines, only fills the gaps in lower-level tools (https://github.com/rust-gamedev/wg/blob/master/charter.md). I'm unsure where along that spectrum we should sit, but I just trying to be wary of coming across in a way that says "hey, this WG has *the* way of doing machine learning in Rust".

xd009642 replied:
Hmm I'd say both working groups are working towards ecosystems just a bit more high level versus low level. What about "Begin to shape the developing rust machine learning ecosystem".

xd009642 replied:
Also I'll endeavour to turn up to the meeting next week as long as I have no clashes :smile:

quietlychris replied:
Hmm, that might work. Let me make the other changes, and then we can maybe take this in particular to the woodshed a little bit more

quietlychris replied:
I've made the other changes, but this is still an area that I think could use some more work

quietlychris committed:
Updates based on discussion; still need to clarify WG goals a little more

bytesnake commented on README.md:
```diff
@@ -5,29 +5,33 @@
 
 The Rust-ML working group is currently at an early stage of development, and is still deciding on several choices at an organizational level. This document will be updated accordingly as those decisions are made.
```
For me the `several choices` is a bit too vague. Perhaps we can word this as 

> The Rust-ML working group is currently at an early stage of development, and its prospect role and function in the wider Rust ecosystem is currently explored. This document will reflect those updates and inform the wider community on decisions made by the Rust-ML group.

I think it is useful to add a general statement, why we believe that Rust and Machine Learning is worthwhile:
> We strongly believe that Rust can contribute to the Machine Learning community with its correctness and speed. It brings the promise of a unified scientific workhorse by combining elegance and performance, previously only found in separate languages. We hope that this will produce libraries, which are easier to use, easier to iterate on and give a standardized level of performance across the ecosystem.

deg4uss3r replied:
:+1: I really like these changes

quietlychris replied:
Me too! I made a couple of edits (mostly grammar) to try to make it flow a little more, but hopefully it still communicates what you wanted. If not, we can change it back

bytesnake commented on README.md:
```diff
@@ -5,29 +5,33 @@
+In general, the Rust-ML working group's goals are to:
 
-The Rust-ML group has a Zulip channel. Feel free to take a look and say hi!
+- Create spaces where people interested in doing machine learning in Rust can meet, ask questions, and discuss their work
```
Perhaps also:
 > Communicate interest in language features, like [GAT](https://github.com/rust-lang/rust/issues/44265), [const generics](https://github.com/rust-lang/rust/issues/44580), f16 types, which are important for easy-to-use machine learning libraries to the language team

quietlychris replied:
I'm a little hesitant to add this in unless there has been some direct contributions/advocacy for these languages features by members of the working group. Do you freely differently?

bytesnake replied:
no, there is no example of direct contributions to the language (at least what I'm aware of)

bytesnake commented on README.md:
```diff
@@ -5,29 +5,33 @@
+
+The Rust-ML group has a Zulip chat. Feel free to take a look and say hi!
 - https://rust-ml.zulipchat.com
 
```
or if you already written something ask for a code review! :)

quietlychris replied:
Good point! It probably makes sense to have that explicitly laid out

bytesnake reviewed (commented):
I think that the intentions are clearly stated, I added something to the beginning and an idea for `goals`. (beware that I'm not a native speaker :sweat_smile:)

Commit:
Added some bytesnake suggestions, cleaned up spelling

quietlychris committed:
Edits some grammar, moves towards active voice

Also just working on this a little bit more. I tried to edit bytesnake's additions, hopefully it's still what he meant to communicate. If not, of course feel free to change back.

quietlychris commented:
Based on the comments, I'm going to merge these initial changes, with the understanding that there will probably be some revision moving forwards
